### PR TITLE
fix(gatsby-remark-autolink-headers): remove accents from anchors

### PR DESCRIPTION
**Pull request by windkomo**
Removing accent characters from URLs (and thus anchors links) seems preferable to me. 
What do you guys think?

### CODE REVIEW

**Comment by pieh** [comment]
This is potentially breaking change (links would potentially change). It's not a blocker - just documenting that we would need bump major version for this change.

I feel like change like this warrant adding some tests to cover cases of headings with accent characters to make sure we don't introduce regressions later on.

**Comment by pieh** [comment]
I'm not sure on this, but are accented characters invalid as element IDs / link targets? If it's not invalid maybe instead of breaking change we could add plugin option to convert to latin (or whatever @sindresorhus/slugify does)?

**Comment by pieh** [comment]
So did some code reading and it seems we can add option to optionally use https://lodash.com/docs/4.17.11#deburr to remove accents

File: packages/gatsby-remark-autolink-headers/src/index.js [hunk 1]
```diff
@@ -16,10 +16,10 @@ module.exports = (
   { markdownAST },
   { icon = svgIcon, className = `anchor`, maintainCase = false }
 ) => {
-  slugs.reset()
```
[the remark this answers is not in the document]
Comment (windkomo) [reply packages/gatsby-remark-autolink-headers/src/index.js]: I think it's the reverse, `slugs.reset` resets the duplicate IDs counter, so slugs are always the same.

**Comment by windkomo** [comment]
@pieh thanks for taking a look!
Accented characters are not invalid in URLs, but to me they are likely to cause issues like encoding.
Slugifying strings should always remove diacritics to avoid any kind of encoding issue so I was very surprised to see accented characters in my address bar 😄

File: packages/gatsby-remark-autolink-headers/src/index.js (hunk 1 above)
[the remark this answers is not in the document]
Comment (pieh) [reply packages/gatsby-remark-autolink-headers/src/index.js]: this reset slugs once for each markdown page so (I think) what it does it makes sure that within single page there won't be duplicated ids if someone create same headers - ie:
```
# Title

## Option `X`

### Example

## Option `Y`

### Example
```
this should make sure that second `Example` header doesn't have same id as first one.

[the remark this answers is not in the document]
Comment (windkomo) [reply packages/gatsby-remark-autolink-headers/src/index.js]: Oh yeah you're right, it resets IDs for each page!

**Comment by windkomo** [comment]
@pieh I think your solution to keep the github slugger thing, and optionally use lodash deburr to remove accents is good. I still think removing accents should be the default behavior, what do you think about introducing this as a feature for now and as a breaking change if we want to later?

**Comment by pieh** [comment]
> what do you think about introducing this as a feature for now and as a breaking change if we want to later?

This sounds perfect to me! We just don't have enough data to make informed decision for all the people, but at least providing option would unblock people that definitely want this behaviour.

**Comment by windkomo** [comment]
@pieh I made some changes, let me know if there's anything wrong!

**Pull request merged**